### ISSUE TITLE
yolo_v3.cfg

### ISSUE THREAD

**Issue by cowarder** [andy-yun/pytorch-0.4-yolov3#56]
Initial yolo_v3.cfg file is for coco, if I use voc dataset to train yolov3, command like this:

%run train.py -d cfg/voc.data -c cfg/yolo_v3.cfg -w yolov3.weights

but voc 2007 dataset has only 20 classes, if I only change the num_classes in cfg file, script doesn't work, could you help to solve this?  thanks!

**Comment by andy-yun** [comment]
@cowarder 
You should modify the filters in the previous layer of yolo.
if you set classes to 20, the filters should be 75.

**Comment by cowarder** [comment]
> @cowarder
> You should modify the filters in the previous layer of yolo.
> if you set classes to 20, the filters should be 75.
Thanks for replying, I did as you said, but during training net, the nPP value is always zero, any reason?

**Comment by cowarder** [comment]
Thanks for replying, I did as you said, but during training net, the nPP value is always zero, any reason?

**Comment by cowarder** [comment]
![image](https://user-images.githubusercontent.com/23216574/55286356-63d7d100-53cd-11e9-96cd-ecfb84c8c7b2.png)
And it crashed during training process....

**Comment by cowarder** [comment]
![image](https://user-images.githubusercontent.com/23216574/55286978-9259aa00-53d5-11e9-90ec-03502de3f012.png)
Compared with other yololayers, the box loss of the first yolo layer would occasionally have very large value, then code crashed.

**Comment by andy-yun** [comment]
@cowarder Refer https://github.com/andy-yun/pytorch-0.4-yolov3/issues/44
same problem. At the beginning of training step, there might be unstable confidence. Thus, we could avoid the nan error by referred trick.

**Comment by cowarder** [comment]
I debugged this. The reason seems to be that the model predicts a large value of P_w and P_h. after it was applied an exponential function as described in 2.1 of [the paper](https://pjreddie.com/media/files/papers/YOLOv3.pdf), the value becomes larger, even get infinity value. So the iou between the target and ground truth becomes nan, then loss gets nan too. 
But I am still wondering why voc would arise this kind of problem, while coco wouldn't?

**Comment by andy-yun** [comment]
@cowarder The training is not stable in voc dataset at the beginning. The initial weight (bootstrap) is taken from the ImageNet data set. The initial weights are therefore far from the voc data set. In my opinion, the distribution of coco data set is not far from the initial weights.

**Comment by cowarder** [comment]
@andy-yun I am still wondering what's the function of correct_yolo_boxes in image.py? And why 'xo' and 'yo' will be divided by 2*net_w and 2*net_h
Thanks for kind replying!
